src/indra_cogex/apps/query_web_app/__init__.py

```diff
@@ -0,0 +1,331 @@
+# -*- coding: utf-8 -*-
+
+"""An app wrapping the query module of indra_cogex."""
+import logging
+from typing import Tuple, Union
+
+import flask
+from flask import request, abort, Response, jsonify
+from more_click import make_web_command
+
+from indra_cogex.client.neo4j_client import Neo4jClient
+from indra_cogex.client.queries import *
+
+app = flask.Flask(__name__)
+
+client = Neo4jClient()
+
+
+logger = logging.getLogger(__name__)
+
+Tup = Tuple[str, str]
+TupOfTups = Tuple[Tup, ...]
+
+
+def _post_request_preproc(*keys) -> Union[TupOfTups, Tup, str, int]:
+    if request.json is None:
+        abort(Response("Missing application/json header.", 415))
+    tups = []
+    for key in keys:
+        if key not in request.json:
+            abort(Response("Parameter '%s' not provided" % key, 415))
+        tups.append(tuple(request.json[key]))
```

Comment:
maybe cleaner to use `request.args.get()`? I think this exposes the path and parameters together

Reply:
Good point, definitely worth looking into to reduce code replication.

Reply:
Although, if using an endpoint like `/<query_func>` to match all available functions, then we can't have the REST API docs for each endpoint listed..

Reply:
I tried providing a set of allowed values for a `/{func_name}` type endpoint in FastAPI, but that didn't get pretty
![image](https://user-images.githubusercontent.com/4621351/155211674-86548222-ab1f-47f6-b599-7d1ab5d3de78.png)

Reply:
you could potentially use a for loop to programatically generate different endpoints

Reply:
Something like this
```python
from fastapi import FastAPI
from typing import Dict, List
from indra_cogex.client import queries


app = FastAPI()


for query_func in queries.__all__:
    @app.post(f"/{query_func}", response_model=List[Dict])
    async def f(term: str):
        return [{"term": term}]
```
making this:
![image](https://user-images.githubusercontent.com/4621351/155327210-efdb2e72-5dc3-45c6-8983-836d961f4fb2.png)

The more verbose part of this would be to make each endpoint unique to its the parameters and theis types as well as the return type.

Reply:
time to break out the `inspect` library!